lib/shelley/src/Cardano/Wallet/Shelley/Launch.hs

```diff
@@ -452,49 +549,76 @@ withBFTNode
     -> FilePath
     -- ^ Parent state directory. Node data will be created in a subdirectory of
     -- this.
+    -> Int
+    -- ^ Which node (assumes key files are indexed by this number)
     -> NodeParams
     -- ^ Parameters used to generate config files.
     -> (FilePath -> Block -> (NetworkParameters, NodeVersionData) -> IO a)
     -- ^ Callback function with genesis parameters
     -> IO a
-withBFTNode tr baseDir (NodeParams severity systemStart (port, peers)) action =
+withBFTNode
+    tr
+    baseDir
+    i
+    (NodeParams severity systemStart mode (port, peers))
+    action =
     bracketTracer' tr "withBFTNode" $ do
-        createDirectory dir
+        createDirectoryIfMissing False dir
+
+        [bftKey, bftDelegCert] <- forM
+            ["delegate-keys.00" <> show i <> ".key", "delegation-cert.00" <> show i <> ".json"]
+            (\f -> copyFile (source </> f) (dir </> f) $> (dir </> f))
 
+        -- TODO: Doesn't support indexing by i
         [vrfPrv, kesPrv, opCert] <- forM
-            ["bft-leader.vrf.skey", "bft-leader.kes.skey", "bft-leader.opcert"]
+            [ "bft-leader" <> ".vrf.skey"
+            , "bft-leader" <> ".kes.skey"
+            , "bft-leader" <> ".opcert"
+            ]
             (\f -> copyFile (source </> f) (dir </> f) $> (dir </> f))
 
         (config, block0, networkParams, versionData)
-            <- genConfig dir severity systemStart
+            <- genConfig dir severity systemStart mode
         topology <- genTopology dir peers
 
-        let cfg = CardanoNodeConfig
+
+        let byronKeys x =
+                    x { nodeDlgCertFile = Just bftDelegCert
+                      , nodeSignKeyFile = Just bftKey
+                      }
+        let shelleyKeys x =
+                    x { nodeOpCertFile = Just opCert
+                      , nodeKesKeyFile = Just kesPrv
+                      , nodeVrfKeyFile = Just vrfPrv
+                      }
+
+        let makeModeSpecific x = case mode of
+                 RunByronShelley _ -> byronKeys . shelleyKeys $ x
+                 RunByron -> byronKeys x
+                 RunShelley -> shelleyKeys x
```

Comment:
Any particular reason to try supporting all three modes? Wouldn't it be easier to just support the Byron/Shelley ?

Reply:
I hoped it could let us still run the integration tests using a pure shelley chain, but that didn't work. So yeah, we can remove it.